evm/vm/block_info.py

```diff
@@ -0,0 +1,62 @@
+class BlockInfo():
+    # For shard chain, refer to the shard coinbase.
+    _coinbase = None
+
+    # For shard chian, block info of period_start_prevhash.
+    _timestamp = None
+    _number = None
+    _difficulty = None
+    _gas_limit = None
+    _prev_headers = None
+
+    def __init__(
+            self,
+            coinbase,
+            timestamp,
+            block_number,
+            difficulty,
+            gas_limit,
+            prev_headers):
+        self._coinbase = coinbase
+        self._timestamp = timestamp
+        self._block_number = block_number
+        self._difficulty = difficulty
+        self._gas_limit = gas_limit
+        self._prev_headers = prev_headers
+
+    @classmethod
+    def generate_by_block_header(cls, block_header, prev_headers, coinbase=None):
```

Comment:
This would closer align with existing naming conventions if it was `generate_from_block_header` or more concisely, just `from_block_header`

Reply:
My carelessness, thank you.